### PR TITLE
Feature: mongodb

### PR DESCRIPTION
**Pull request by R-Mielamud**

### CODE REVIEW

File: backend/src/Infrastructure/Repositories/Abstractions/ReadRepository.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Dapper;
+using Domain.Interfaces;
+using Domain.Common;
+using Infrastructure.Dapper.Interfaces;
+
+namespace Infrastructure.Repositories.Abstractions
+{
+    public class ReadRepository<T> : IReadRepository<T>
```
Comment (anton-matsyshyn) [inline backend/src/Infrastructure/Repositories/Abstractions/ReadRepository.cs:11]: Nice idea!

File: backend/src/Infrastructure/EF/MongoDbContext.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+using System;
+using MongoDB.Driver;
+using Domain.Common;
+using Domain.Entities.Mongo;
+
+namespace Infrastructure.EF
+{
+    public class MongoDbContext
```
Comment (anton-matsyshyn) [inline backend/src/Infrastructure/EF/MongoDbContext.cs:8]: I don't think we need this. Implement MongoConnectionFactory instead

File: backend/src/Domain/Interfaces/IMongoReadRepository.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using Domain.Common;
+
+namespace Domain.Interfaces
+{
+    public interface IMongoReadRepository<T> where T : MongoEntity
```
Comment (anton-matsyshyn) [inline backend/src/Domain/Interfaces/IMongoReadRepository.cs:8]: The key idea to wrap everything with a repository is to have same repository interface everywhere. Check whether it's possible to implement same logic with string id, and if yes, correct IReadRepository and it's implementations.

File: backend/src/Infrastructure/EF/MongoDbContext.cs (hunk 2 above)
Comment (anton-matsyshyn) [reply backend/src/Infrastructure/EF/MongoDbContext.cs:8]: And it's better to do this in Mongo/ folder, not EF/

File: backend/src/Infrastructure/Mongo/MongoConnectionFactory.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        public IMongoCollection<ApplicantCv> ApplicantCvs { get; set; }
+
+        public MongoConnectionFactory()
+        {
+            _client = new MongoClient(_connectionUri);
+            _connection = _client.GetDatabase(_database);
+            ApplicantCvs = _connection.GetCollection<ApplicantCv>(nameof(ApplicantCv));
+        }
+
+        public IMongoCollection<T> Collection<T>()
```
Comment (anton-matsyshyn) [inline backend/src/Infrastructure/Mongo/MongoConnectionFactory.cs:24]: I also see no need in storing collection in ConnectionFactory, better do it in repository

File: backend/src/Infrastructure/InfrastructureDependencyInjection.cs [hunk 5]
```diff
@@ -38,11 +37,13 @@ private static IServiceCollection AddDatabaseContext(this IServiceCollection ser
...
-                            b => b.MigrationsAssembly(typeof(ApplicationDbContext).Assembly.FullName)
-                        )
-                );
+                options => options.UseSqlServer(
+                    connectionString,
+                    b => b.MigrationsAssembly(typeof(ApplicationDbContext).Assembly.FullName)
+                )
+            );
+
+            services.AddSingleton<MongoConnectionFactory>();
```
Comment (anton-matsyshyn) [inline backend/src/Infrastructure/InfrastructureDependencyInjection.cs:46]: Transient should be fine

**Comment by anton-matsyshyn** [comment]
Add guid generation to CreateEntityCommandHandler

File: backend/src/Infrastructure/InfrastructureDependencyInjection.cs (hunk 5 above)
Comment (R-Mielamud) [reply backend/src/Infrastructure/InfrastructureDependencyInjection.cs:46]: Transient will reconnect to database every time we use it.

Comment (anton-matsyshyn) [reply backend/src/Infrastructure/InfrastructureDependencyInjection.cs:46]: In such case implement method, which initialize connection and returns it

Comment (anton-matsyshyn) [reply backend/src/Infrastructure/InfrastructureDependencyInjection.cs:46]: So your connection factory should contains method GetConnection, which returns new connection to mongo db

Comment (anton-matsyshyn) [reply backend/src/Infrastructure/InfrastructureDependencyInjection.cs:46]: And it also would be great if you added IMongoConnectionFactory, just like with connection to sql server

**Comment by anton-matsyshyn** [comment]
IMongoConnectionFactory and MongoConnectionFactory should be placed in Mongo/Interfaces and Mongo/Services folders

**Pull request merged**